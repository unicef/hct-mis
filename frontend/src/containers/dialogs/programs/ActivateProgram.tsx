import React, { useState } from 'react';
import styled from 'styled-components';
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  Typography,
} from '@material-ui/core';
import {
  ProgramNode,
  ProgramStatus,
  useUpdateProgramMutation,
} from '../../../__generated__/graphql';
import { PROGRAM_QUERY } from '../../../apollo/queries/Program';

const DialogTitleWrapper = styled.div`
  border-bottom: 1px solid ${({theme})=>theme.hctPalette.lighterGray};
`;

const DialogFooter = styled.div`
  padding: 12px 16px;
  margin: 0;
  border-top: 1px solid ${({theme})=>theme.hctPalette.lighterGray};
  text-align: right;
`;

const DialogDescription = styled.div`
  margin: 20px 0;
  font-size: 14px;
  color: rgba(0, 0, 0, 0.54);
`;

interface ActivateProgramProps {
  program: ProgramNode;
}

export function ActivateProgram({
  program,
}: ActivateProgramProps): React.ReactElement {
  const [open, setOpen] = useState(false);
  const [mutate] = useUpdateProgramMutation({
    update(cache, { data: { updateProgram } }) {
      cache.writeQuery({
        query: PROGRAM_QUERY,
        variables: {
          id: program.id,
        },
        data: { program: updateProgram.program },
      });
    },
  });
  const activateProgram = async (): Promise<void> => {
    await mutate({
      variables: {
        programData: {
          id: program.id,
          status: ProgramStatus.Active,
        },
      },
    });
    setOpen(false);
  };
  return (
    <span>
      <Button variant='contained' color='primary' onClick={() => setOpen(true)}>
        Activate
      </Button>
      <Dialog
        open={open}
        onClose={() => setOpen(false)}
        scroll='paper'
        aria-labelledby='form-dialog-title'
      >
        <DialogTitleWrapper>
          <DialogTitle id='scroll-dialog-title'>
            <Typography variant='h6'>Activate Programme</Typography>
          </DialogTitle>
        </DialogTitleWrapper>
        <DialogContent>
          <DialogDescription>
            Are you sure you want to activate this Programme and push data to
            CashAssist?
          </DialogDescription>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button onClick={() => setOpen(false)} color='primary'>
              CANCEL
            </Button>
            <Button
              type='submit'
              color='primary'
              variant='contained'
              onClick={activateProgram}
            >
              ACTIVATE
            </Button>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </span>
  );
}
