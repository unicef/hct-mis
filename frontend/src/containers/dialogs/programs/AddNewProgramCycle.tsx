import {
  Box,
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  Grid,
} from '@material-ui/core';
import CalendarTodayRoundedIcon from '@material-ui/icons/CalendarTodayRounded';
import { Add } from '@material-ui/icons';
import { Field, Formik } from 'formik';
import moment from 'moment';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import * as Yup from 'yup';
import {
  AllProgramsQuery,
  ProgramNode,
  ProgramStatus,
  useUpdateProgramMutation,
} from '../../../__generated__/graphql';
import { ALL_PROGRAMS_QUERY } from '../../../apollo/queries/program/AllPrograms';
import { PROGRAM_QUERY } from '../../../apollo/queries/program/Program';
import { AutoSubmitFormOnEnter } from '../../../components/core/AutoSubmitFormOnEnter';
import { GreyText } from '../../../components/core/GreyText';
import { LabelizedField } from '../../../components/core/LabelizedField';
import { LoadingButton } from '../../../components/core/LoadingButton';
import { Missing } from '../../../components/core/Missing';
import { DialogDescription } from '../DialogDescription';
import { DialogFooter } from '../DialogFooter';
import { DialogTitleWrapper } from '../DialogTitleWrapper';
import { useBaseUrl } from '../../../hooks/useBaseUrl';
import { useSnackbar } from '../../../hooks/useSnackBar';
import { FormikDateField } from '../../../shared/Formik/FormikDateField';
import { programCompare, today } from '../../../utils/utils';
import { FormikTextField } from '../../../shared/Formik/FormikTextField';

interface AddNewProgramCycleProps {
  program: ProgramNode;
}

export const AddNewProgramCycle = ({
  program,
}: AddNewProgramCycleProps): React.ReactElement => {
  const { t } = useTranslation();
  const [open, setOpen] = useState(false);
  const { showMessage } = useSnackbar();
  const { baseUrl, businessArea } = useBaseUrl();
  const [step, setStep] = useState(1);

  const [mutate, { loading }] = useUpdateProgramMutation({
    update(cache, { data: { updateProgram } }) {
      cache.writeQuery({
        query: PROGRAM_QUERY,
        variables: {
          id: program.id,
        },
        data: { program: updateProgram.program },
      });
      const allProgramsData: AllProgramsQuery = cache.readQuery({
        query: ALL_PROGRAMS_QUERY,
        variables: { businessArea },
      });
      allProgramsData.allPrograms.edges.sort(programCompare);
      cache.writeQuery({
        query: ALL_PROGRAMS_QUERY,
        variables: { businessArea },
        data: allProgramsData,
      });
    },
  });

  //es-lint-disable-next-line
  const save = async (): Promise<void> => {
    const response = await mutate({
      variables: {
        programData: {
          id: program.id,
          status: ProgramStatus.Active,
        },
        version: program.version,
      },
    });
    if (!response.errors && response.data.updateProgram) {
      showMessage(t('Programme activated.'), {
        pathname: `/${baseUrl}/details/${response.data.updateProgram.program.id}`,
        dataCy: 'snackbar-program-activate-success',
      });
      setOpen(false);
    } else {
      showMessage(t('Programme activate action failed.'), {
        dataCy: 'snackbar-program-activate-failure',
      });
    }
  };

  const validationSchema = Yup.object().shape({
    title: Yup.string()
      .required(t('Programme Cycle title is required'))
      .min(2, t('Too short'))
      .max(150, t('Too long')),
    newProgramCycleStartDate: Yup.date().required(t('Start Date is required')),
    newProgramCycleEndDate: Yup.date()
      .required(t('End Date is required'))
      .min(today, t('End Date cannot be in the past'))
      .when(
        'newProgramCycleStartDate',
        (newProgramCycleStartDate, schema) =>
          newProgramCycleStartDate &&
          schema.min(
            newProgramCycleStartDate,
            `${t('End date have to be greater than')} ${moment(
              newProgramCycleStartDate,
            ).format('YYYY-MM-DD')}`,
          ),
        '',
      ),
    existingProgramCycleEndDate: Yup.date()
      .required(t('End Date is required'))
      .min(today, t('End Date cannot be in the past'))
      .when(
        'startDate',
        (startDate, schema) =>
          startDate &&
          schema.min(
            startDate,
            `${t('End date have to be greater than')} ${moment(
              startDate,
            ).format('YYYY-MM-DD')}`,
          ),
        '',
      ),
  });

  const initialValues: {
    [key: string]: string | boolean | number;
  } = {
    title: '',
    existingProgramCycleStartDate: '',
    newProgramCycleStartDate: '',
    newProgramCycleEndDate: '',
  };

  return (
    <>
      <Button
        startIcon={<Add />}
        variant='outlined'
        color='primary'
        data-cy='button-add-new-program-cycle'
        onClick={() => setOpen(true)}
      >
        {t('Add New Programme Cycle')}
      </Button>
      <Dialog open={open} onClose={() => setOpen(false)} scroll='paper'>
        <Formik
          initialValues={initialValues}
          onSubmit={(values) => {
            console.log(values);
          }}
          validationSchema={validationSchema}
        >
          {({ submitForm }) => (
            <>
              {open && <AutoSubmitFormOnEnter />}
              <DialogTitleWrapper>
                <DialogTitle>
                  <Box display='flex' justifyContent='space-between'>
                    <Box>{t('Add New Programme Cycle')}</Box>
                    <Box>{step === 1 ? '1/2' : '2/2'}</Box>
                  </Box>
                </DialogTitle>
              </DialogTitleWrapper>
              <DialogContent>
                <DialogDescription>
                  <GreyText>
                    {step === 1
                      ? t(
                          'Before you create a new cycle, it is necessary to specify the end date of the existing cycle',
                        )
                      : t('Enter data for the new Programme Cycle')}
                  </GreyText>
                </DialogDescription>
                {step === 1 ? (
                  <Grid container spacing={3}>
                    <Grid item xs={6}>
                      <LabelizedField label={t('Programme Cycle Title')}>
                        {t('Default Programme Cycle')}
                      </LabelizedField>
                    </Grid>
                    <Grid item xs={6}>
                      <LabelizedField label={t('Start Date')}>
                        <Missing />
                      </LabelizedField>
                    </Grid>
                    <Grid item xs={6}>
                      <Field
                        name='existingProgramCycleEndDate'
                        label={t('End Date')}
                        component={FormikDateField}
                        required
                        fullWidth
                        decoratorEnd={
                          <CalendarTodayRoundedIcon color='disabled' />
                        }
                        data-cy='input-existing-end-date'
                      />
                    </Grid>
                  </Grid>
                ) : (
                  <Grid container spacing={3}>
                    <Grid item xs={12}>
                      <Field
                        name='programCycleTitle'
                        fullWidth
                        variant='outlined'
                        label={t('Programme Cycle Title')}
                        component={FormikTextField}
                        required
                      />
                    </Grid>
                    <Grid item xs={6}>
                      <Field
                        name='newProgramCycleStartDate'
                        label={t('Start Date')}
                        component={FormikDateField}
                        required
                        fullWidth
                        decoratorEnd={
                          <CalendarTodayRoundedIcon color='disabled' />
                        }
                        data-cy='input-new-start-date'
                      />
                    </Grid>
                    <Grid item xs={6}>
                      <Field
                        name='newProgramCycleEndDate'
                        label={t('End Date')}
                        component={FormikDateField}
                        fullWidth
                        decoratorEnd={
                          <CalendarTodayRoundedIcon color='disabled' />
                        }
                        data-cy='input-new-end-date'
                      />
                    </Grid>
                  </Grid>
                )}
              </DialogContent>
              <DialogFooter>
                <DialogActions>
                  <Button onClick={() => setOpen(false)}>{t('CANCEL')}</Button>
                  {step === 2 && (
                    <LoadingButton
                      loading={loading}
                      type='submit'
                      color='primary'
                      variant='contained'
                      onClick={submitForm}
                      data-cy='button-create'
                    >
                      {t('Create')}
                    </LoadingButton>
                  )}
                  {step === 1 && (
                    <Button
                      color='primary'
                      variant='contained'
                      onClick={() => setStep(2)}
                      data-cy='button-next'
                    >
                      {t('Next')}
                    </Button>
                  )}
                </DialogActions>
              </DialogFooter>
            </>
          )}
        </Formik>
      </Dialog>
    </>
  );
};
