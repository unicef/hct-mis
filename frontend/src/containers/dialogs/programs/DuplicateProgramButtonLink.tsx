import { IconButton } from '@material-ui/core';
import { FileCopy } from '@material-ui/icons';
import React, { ReactElement } from 'react';
import { Link } from 'react-router-dom';
import { ProgramQuery } from '../../../__generated__/graphql';
import { useBaseUrl } from '../../../hooks/useBaseUrl';

interface DuplicateProgramButtonLinkProps {
  program: ProgramQuery['program'];
}

export const DuplicateProgramButtonLink = ({
  program,
}: DuplicateProgramButtonLinkProps): ReactElement => {
  const { baseUrl } = useBaseUrl();

  return (
    <IconButton
      component={Link}
      to={`/${baseUrl}/duplicate/${program.id}`}
      data-cy='button-copy-program'
    >
      <FileCopy />
    </IconButton>
  );
};
