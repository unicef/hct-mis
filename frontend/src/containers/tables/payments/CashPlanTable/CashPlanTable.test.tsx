import { MockedProvider } from '@apollo/react-testing';
import React from 'react';
import { act } from '@testing-library/react';
import wait from 'waait';
import { CashPlanTable } from '.';
import { render, ApolloLoadingLink } from '../../../../testUtils/testUtils';
import { fakeApolloAllCashPlans } from '../../../../../fixtures/payments/fakeApolloAllCashPlans';
import { fakeProgram } from '../../../../../fixtures/programs/fakeProgram';

describe('containers/tables/payments/CashPlanTable', () => {
  it('should render with data', async () => {
    const { container } = render(
      <MockedProvider addTypename={false} mocks={fakeApolloAllCashPlans}>
        <CashPlanTable program={fakeProgram} />
      </MockedProvider>,
    );
    await act(() => wait(0)); // wait for response

    expect(container).toMatchSnapshot();
  });

  it('should render loading', () => {
    const { container } = render(
      <MockedProvider
        link={new ApolloLoadingLink()}
        addTypename={false}
        mocks={fakeApolloAllCashPlans}
      >
        <CashPlanTable program={fakeProgram} />
      </MockedProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
