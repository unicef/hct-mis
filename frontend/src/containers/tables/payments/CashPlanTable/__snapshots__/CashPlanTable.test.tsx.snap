// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/tables/payments/CashPlanTable should render loading 1`] = `
<div>
  <div
    class="sc-bdVaJa iMXEqU"
    data-cy="loading-container"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`containers/tables/payments/CashPlanTable should render with data 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiTableContainer-root"
      >
        <div
          class="MuiBox-root MuiBox-root-7"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular makeStyles-root-8 MuiToolbar-gutters"
          >
            <h6
              class="MuiTypography-root makeStyles-title-10 MuiTypography-h6"
              data-cy="table-title"
            >
              Payment Plans
            </h6>
          </div>
          <div
            class="MuiBox-root MuiBox-root-11"
          />
        </div>
        <table
          aria-label="enhanced table"
          aria-labelledby="tableTitle"
          class="MuiTable-root makeStyles-table-3"
          data-cy="table-title"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                scope="col"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiTableSortLabel-root sc-EHOje dirFCH"
                  data-cy="table-label"
                  role="button"
                  tabindex="0"
                >
                  Payment Plan ID
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                scope="col"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiTableSortLabel-root sc-EHOje dirFCH"
                  data-cy="table-label"
                  role="button"
                  tabindex="0"
                >
                  Status
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiTableSortLabel-root sc-EHOje dirFCH"
                  data-cy="table-label"
                  role="button"
                  tabindex="0"
                >
                  Num. of Households
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                scope="col"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiTableSortLabel-root sc-EHOje dirFCH"
                  data-cy="table-label"
                  role="button"
                  tabindex="0"
                >
                  Currency
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiTableSortLabel-root sc-EHOje dirFCH"
                  data-cy="table-label"
                  role="button"
                  tabindex="0"
                >
                  Total Entitled Quantity
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiTableSortLabel-root sc-EHOje dirFCH"
                  data-cy="table-label"
                  role="button"
                  tabindex="0"
                >
                  Total Delivered Quantity
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                scope="col"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiTableSortLabel-root sc-EHOje dirFCH"
                  data-cy="table-label"
                  role="button"
                  tabindex="0"
                >
                  Total Undelivered Quantity
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                scope="col"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiTableSortLabel-root sc-EHOje dirFCH"
                  data-cy="table-label"
                  role="button"
                  tabindex="0"
                >
                  Dispersion Date
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root sc-gzVnrw btWNie MuiTableRow-hover"
              data-cy="cash-plan-table-row"
              role="checkbox"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <a
                  class="sc-iwsKbI kdOUDe"
                  href="/afghanistan/programs/UHJvZ3JhbU5vZGU6YzRkNTY1N2QtMWEyOS00NmUxLTgxOTAtZGY3Zjg1YTBkMmVm/payment-module/payment-plans/UGF5bWVudFBsYW5Ob2RlOmQ4NDdkZDFlLTkwYmYtNDQzZC1iNWI1LWNiZTcwMDQ5YzYyMA=="
                >
                  <div
                    style="text-overflow: ellipsis;"
                  >
                    PP-0060-23-00000017
                  </div>
                </a>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  class="sc-htoDjs koRFFa"
                >
                  <div
                    class="sc-dnqmqq QDGnp"
                    data-cy="status-container"
                  >
                    PENDING
                  </div>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2,200.00
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                1,000.00
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                1,200.00
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                -
              </td>
            </tr>
            <tr
              class="MuiTableRow-root sc-gzVnrw btWNie MuiTableRow-hover"
              data-cy="cash-plan-table-row"
              role="checkbox"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <a
                  class="sc-iwsKbI kdOUDe"
                  href="/afghanistan/programs/UHJvZ3JhbU5vZGU6YzRkNTY1N2QtMWEyOS00NmUxLTgxOTAtZGY3Zjg1YTBkMmVm/payment-module/payment-plans/UGF5bWVudFBsYW5Ob2RlOjQ4YmU4NzVmLWUxYzMtNDM3OS1iMzc1LTgwMzlkYWU4ODQ5MQ=="
                >
                  <div
                    style="text-overflow: ellipsis;"
                  >
                    PP-0060-23-00000015
                  </div>
                </a>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  class="sc-htoDjs koRFFa"
                >
                  <div
                    class="sc-dnqmqq QDGnp"
                    data-cy="status-container"
                  >
                    PENDING
                  </div>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2,800.00
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2,800.00
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                0.00
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                -
              </td>
            </tr>
            <tr
              class="MuiTableRow-root sc-gzVnrw btWNie MuiTableRow-hover"
              data-cy="cash-plan-table-row"
              role="checkbox"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <a
                  class="sc-iwsKbI kdOUDe"
                  href="/afghanistan/programs/UHJvZ3JhbU5vZGU6YzRkNTY1N2QtMWEyOS00NmUxLTgxOTAtZGY3Zjg1YTBkMmVm/payment-module/payment-plans/UGF5bWVudFBsYW5Ob2RlOjBkNDVlNjQzLTY1YWMtNDIzOS04ZmI5LWIyNzRhNjlmMmE2ZA=="
                >
                  <div
                    style="text-overflow: ellipsis;"
                  >
                    PP-0060-23-00000005
                  </div>
                </a>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  class="sc-htoDjs koRFFa"
                >
                  <div
                    class="sc-dnqmqq QDGnp"
                    data-cy="status-container"
                  >
                    ACTIVE
                  </div>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                200.00
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                150.00
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                50.00
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                -
              </td>
            </tr>
            <tr
              class="MuiTableRow-root sc-gzVnrw btWNie MuiTableRow-hover"
              data-cy="cash-plan-table-row"
              role="checkbox"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <a
                  class="sc-iwsKbI kdOUDe"
                  href="/afghanistan/programs/UHJvZ3JhbU5vZGU6YzRkNTY1N2QtMWEyOS00NmUxLTgxOTAtZGY3Zjg1YTBkMmVm/payment-module/payment-plans/UGF5bWVudFBsYW5Ob2RlOjc3MDQ1YmUwLTUzOTUtNDdlNy1hZWZiLTYwZjNjNDgxNzE5Zg=="
                >
                  <div
                    style="text-overflow: ellipsis;"
                  >
                    PP-0060-23-00000001
                  </div>
                </a>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  class="sc-htoDjs koRFFa"
                >
                  <div
                    class="sc-dnqmqq QDGnp"
                    data-cy="status-container"
                  >
                    ACTIVE
                  </div>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                700.00
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                600.00
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                100.00
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                -
              </td>
            </tr>
            <tr
              class="MuiTableRow-root sc-gzVnrw btWNie MuiTableRow-hover"
              data-cy="cash-plan-table-row"
              role="checkbox"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <a
                  class="sc-iwsKbI kdOUDe"
                  href="/afghanistan/programs/UHJvZ3JhbU5vZGU6YzRkNTY1N2QtMWEyOS00NmUxLTgxOTAtZGY3Zjg1YTBkMmVm/cashplans/Q2FzaFBsYW5Ob2RlOjk1NGY3OWE3LTdiZDYtNDU5ZS05ZGE4LTQzNjg3ZDY0NjYyNg=="
                >
                  <div
                    style="text-overflow: ellipsis;"
                  >
                    123-21-CSH-00003
                  </div>
                </a>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                <div
                  class="sc-htoDjs koRFFa"
                >
                  <div
                    class="sc-dnqmqq QDGnp"
                    data-cy="status-container"
                  >
                    PENDING
                  </div>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                5
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                2,463,289.03
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                14,923,041.29
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                16,301,871.34
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
              >
                -
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="MuiTablePagination-root"
        data-cy="table-pagination"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
        >
          <div
            class="MuiTablePagination-spacer"
          />
          <p
            class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
            id="mui-5806"
          >
            Rows per page:
          </p>
          <div
            class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="mui-5806 mui-73230"
              class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
              id="mui-73230"
              role="button"
              tabindex="0"
            >
              5
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="5"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          >
            1-5 of 7
          </p>
          <div
            class="MuiTablePagination-actions"
          >
            <button
              aria-label="Previous page"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              title="Previous page"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Next page"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              tabindex="0"
              title="Next page"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
